src/app/_components/NextAuth/SignIn.tsx

```diff
@@ -0,0 +1,27 @@
+"use client";
+
+import { useSession } from "next-auth/react";
+import { useRouter } from 'next/navigation';
+import { signIn } from 'next-auth/react';
+
+// Page for signing in
+const SignIn = () => {
+  const { data: session, status } = useSession();
```

Comment:
No need to make the change now, but in the future we can have all authentication done server side using GetServerSideProps

Reply:
[Let's look into this](https://create.t3.gg/en/usage/next-auth#retrieving-session-server-side), it provides security benefits and better client side performance.